src/toil/wdl/wdltoil.py

```diff
@@ -3322,125 +3324,132 @@ def main() -> None:
 
     # Get the execution directory
     execution_dir = os.getcwd()
-
-    with Toil(options) as toil:
-        if options.restart:
-            output_bindings = toil.restart()
-        else:
-            # Load the WDL document
-            document: WDL.Tree.Document = WDL.load(options.wdl_uri, read_source=toil_read_source)
-
-            # See if we're going to run a workflow or a task
-            target: Union[WDL.Tree.Workflow, WDL.Tree.Task]
-            if document.workflow:
-                target = document.workflow
-            elif len(document.tasks) == 1:
-                target = document.tasks[0]
-            elif len(document.tasks) > 1:
-                raise WDL.Error.InputError("Multiple tasks found with no workflow! Either add a workflow or keep one task.")
+    try:
```

Comment:
Having to indent so much code here suggests that at some point we might want to break out subtasks into different functions, so we have less code in this big block.